### PR TITLE
rdiary: init at 0.1.0

### PR DESCRIPTION
**Pull request by lonyelon**
Simple CLI diary writer made in rust.

It is a tool I made and use daily, maybe someone other than me can find use for it.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/rd/rdiary/package.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+rustPlatform.buildRustPackage rec {
+  pname = "rdiary";
+  version = "0-unstable-2024-10-02";
+
+  src = fetchFromGitHub {
+    owner = "lonyelon";
+    repo = "rdiary";
+    rev = "v0.1.0";
+    sha256 = "sha256-N6wyZdo3CL57DzzRx3Rc4iYwlUyrTYPJOaBLVKzSwUw=";
```
Comment (kirillrdy) [inline pkgs/by-name/rd/rdiary/package.nix:16]:
```suggestion
    hash = "sha256-N6wyZdo3CL57DzzRx3Rc4iYwlUyrTYPJOaBLVKzSwUw=";
```

File: pkgs/by-name/rd/rdiary/package.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  };
+
+  cargoSha256 = "sha256-7pbBwyCvvZ+YVybKH8kGD2BId54q95iVMLOWu/K4rUU=";
```
Comment (kirillrdy) [inline pkgs/by-name/rd/rdiary/package.nix:19]:
```suggestion
  cargoHash = "sha256-7pbBwyCvvZ+YVybKH8kGD2BId54q95iVMLOWu/K4rUU=";
```

File: pkgs/by-name/rd/rdiary/package.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  meta = with lib; {
+    description = "Very simple diary writer";
+    homepage = "https://github.com/lonyelon/rdiary";
+    license = licenses.gpl3;
+    maintainers = with maintainers; [ lonyelon ];
+    platforms = platforms.linux;
```
Comment (kirillrdy) [inline pkgs/by-name/rd/rdiary/package.nix:26]: can this be omitted, in which case rust platforms are used ?

File: pkgs/by-name/rd/rdiary/package.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  src = fetchFromGitHub {
+    owner = "lonyelon";
+    repo = "rdiary";
+    rev = "v0.1.0";
+    sha256 = "sha256-N6wyZdo3CL57DzzRx3Rc4iYwlUyrTYPJOaBLVKzSwUw=";
+  };
+
+  cargoSha256 = "sha256-7pbBwyCvvZ+YVybKH8kGD2BId54q95iVMLOWu/K4rUU=";
+
+  meta = with lib; {
```
Comment (kirillrdy) [inline pkgs/by-name/rd/rdiary/package.nix:21]: with lib;  has unexpected side effects https://github.com/NixOS/nixpkgs/issues/208242

File: pkgs/by-name/rd/rdiary/package.nix [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    description = "Very simple diary writer";
+    homepage = "https://github.com/lonyelon/rdiary";
+    license = licenses.gpl3;
```
Comment (kirillrdy) [inline pkgs/by-name/rd/rdiary/package.nix:24]: gpl3 is deprecated consider either gpl3Plus or gpl3Only

File: pkgs/by-name/rd/rdiary/package.nix [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "rdiary";
+  version = "0-unstable-2024-10-02";
+
+  src = fetchFromGitHub {
+    owner = "lonyelon";
+    repo = "rdiary";
+    rev = "v0.1.0";
```
Comment (kirillrdy) [inline pkgs/by-name/rd/rdiary/package.nix:15]: this should either match `version` or be full hash given this is unstable version

**Comment by lonyelon** [comment]
Result of `nixpkgs-review`:

```sh
[pts/1:10240][sergio@icebear:~/proj/nixpkgs](rdiary)$ nix-shell -p nixpkgs-review --run "nixpkgs-review pr 345894"
$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/345894/head:refs/nixpkgs-review/1
Desde https://github.com/NixOS/nixpkgs
 + f4207a796ef4...0a89d2432afe refs/pull/345894/head -> refs/nixpkgs-review/1  (actualización forzada)
$ git worktree add /home/sergio/.cache/nixpkgs-review/pr-345894-5/nixpkgs f207b543e2c3b0c14a461dd3ea50f6af55781db1
Preparando árbol de trabajo (HEAD desacoplado f207b543e2c3)
Actualizando archivos: 100% (43535/43535), listo.
HEAD está ahora en f207b543e2c3 brainflow: 5.13.3 -> 5.14.0 (#345939)
$ nix-env --extra-experimental-features no-url-literals --option system x86_64-linux -f <nixpkgs> --nix-path nixpkgs=/home/sergio/.cache/nixpkgs-review/pr-345894-5/nixpkgs nixpkgs-overlays=/tmp/tmpoj36uhug -qaP --xml --out-path --show-trace --no-allow-import-from-derivation
$ git merge --no-commit --no-ff 0a89d2432afe50cab61b5758b91c3f2fee95f8b5
Fusión automática fue bien; detenida antes del commit como se solicitó
$ nix-env --extra-experimental-features no-url-literals --option system x86_64-linux -f <nixpkgs> --nix-path nixpkgs=/home/sergio/.cache/nixpkgs-review/pr-345894-5/nixpkgs nixpkgs-overlays=/tmp/tmpoj36uhug -qaP --xml --out-path --show-trace --no-allow-import-from-derivation --meta
1 package added:
rdiary (init at 0.1.0)

$ nix build --file /nix/store/wv6yx1iq5v90jcv03c2v5wyyfbz3xpx7-nixpkgs-review-2.10.5/lib/python3.12/site-packages/nixpkgs_review/nix/review-shell.nix --nix-path 'nixpkgs=/home/sergio/.cache/nixpkgs-review/pr-345894-5/nixpkgs nixpkgs-overlays=/tmp/tmpoj36uhug' --extra-experimental-features 'nix-command no-url-literals' --no-link --keep-going --no-allow-import-from-derivation --option build-use-sandbox relaxed --argstr system x86_64-linux --argstr nixpkgs-path /home/sergio/.cache/nixpkgs-review/pr-345894-5/nixpkgs --argstr nixpkgs-config-path /tmp/tmpeufej34c.nix --argstr attrs-path /home/sergio/.cache/nixpkgs-review/pr-345894-5/attrs.nix

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/345894

1 package built:
rdiary

$ /nix/store/1fp0ggfyfgn6ydsgc8sdwqv5n5gn9nal-nix-2.18.8/bin/nix-shell --argstr system x86_64-linux --argstr nixpkgs-path /home/sergio/.cache/nixpkgs-review/pr-345894-5/nixpkgs --argstr nixpkgs-config-path /tmp/tmpeufej34c.nix --argstr attrs-path /home/sergio/.cache/nixpkgs-review/pr-345894-5/attrs.nix --nix-path 'nixpkgs=/home/sergio/.cache/nixpkgs-review/pr-345894-5/nixpkgs nixpkgs-overlays=/tmp/tmpoj36uhug' /nix/store/wv6yx1iq5v90jcv03c2v5wyyfbz3xpx7-nixpkgs-review-2.10.5/lib/python3.12/site-packages/nixpkgs_review/nix/review-shell.nix
```

**Review by FliegendeWurst — COMMENTED** [review]
Package LGTM, but I am not sure whether this tool has enough users to justify inclusion in nixpkgs. Perhaps it could go in the [NUR](https://github.com/nix-community/NUR) first.

**Comment by lonyelon** [comment]
> Package LGTM, but I am not sure whether this tool has enough users to justify inclusion in nixpkgs. Perhaps it could go in the [NUR](https://github.com/nix-community/NUR) first.

Yeah, I will just stick to the flake. Thanks though.

I am closing this.

**Pull request closed without merging**